### PR TITLE
Add --exportViewDefinition option to imodel-transformer

### PR DESCRIPTION
**Pull request by scsewall**

### CODE REVIEW

File: test-apps/imodel-transformer/src/ElementUtils.ts [hunk 1]
```diff
@@ -85,4 +88,49 @@ export namespace ElementUtils {
...
+   */
+  export function insertViewDefinition(iModelDb: IModelDb, name: string, makeDefault?: boolean): Id64String {
+    const definitionModelId = IModel.dictionaryId;
+    const viewCode = ViewDefinition.createCode(iModelDb, definitionModelId, name);
+    let viewId = iModelDb.elements.queryElementIdByCode(viewCode);
+    if (viewId === undefined) {
+      const modelSelectorId = ModelSelector.insert(iModelDb, definitionModelId, name, queryModelIds(iModelDb, SpatialModel.classFullName));
+      const categorySelectorId = CategorySelector.insert(iModelDb, definitionModelId, name, querySpatialCategoryIds(iModelDb));
+      const displayStyleId = DisplayStyle3d.insert(iModelDb, definitionModelId, name);
+      viewId = OrthographicViewDefinition.insert(iModelDb, definitionModelId, name, modelSelectorId, categorySelectorId, displayStyleId, iModelDb.projectExtents, StandardViewIndex.Iso);
```
Comment (pmconne) [inline test-apps/imodel-transformer/src/ElementUtils.ts:106]: Why orthographic??

File: test-apps/imodel-transformer/src/Transformer.ts [hunk 2]
```diff
@@ -78,6 +80,31 @@ export class Transformer extends IModelTransformer {
       this._targetPhysicalModelId = PhysicalModel.insert(this.targetDb, IModel.rootSubjectId, "CombinedPhysicalModel"); // WIP: Id should be passed in, not inserted here
       this.importer.doNotUpdateElementIds.add(this._targetPhysicalModelId);
     }
+    if (options?.exportViewDefinition) {
```
Comment (pmconne) [inline test-apps/imodel-transformer/src/Transformer.ts:83]: I don't suppose you currently have a way to filter geometry from geometry streams based on class (constructions etc)?

**Review by pmconne — COMMENTED** [review]
This appears to ignore the view clip - is that planned for a subsequent PR?

**Comment by scsewall** [comment]
> This appears to ignore the view clip - is that planned for a subsequent PR?

Yes, I was going to do that in a separate PR.

File: test-apps/imodel-transformer/src/Transformer.ts (hunk 2 above)
Comment (scsewall) [reply test-apps/imodel-transformer/src/Transformer.ts:83]: Not yet, but that could be added. I think that would best be done in native code while cloning the GeometryStream (same spot as where entries are filtered by SubCategory). Seems like `Primary` should never be excluded. I can see wanting to exclude `Construction` and `Dimension`. Not sure about `Pattern`. Thoughts?

Comment (pmconne) [reply test-apps/imodel-transformer/src/Transformer.ts:83]: Yes, patterns too. If view flags say not to draw some class of geometry ideally the transformer will exclude geometry of that class.

File: test-apps/imodel-transformer/src/ElementUtils.ts (hunk 1 above)
Comment (scsewall) [reply test-apps/imodel-transformer/src/ElementUtils.ts]: No good reason. Is my latest change better?

Comment (pmconne) [reply test-apps/imodel-transformer/src/ElementUtils.ts]: Yes. OrthographicViewDefinition is almost never what you want, all it does is forbid you from turning camera on.

File: test-apps/imodel-transformer/src/Transformer.ts (hunk 2 above)
Comment (pmconne) [reply test-apps/imodel-transformer/src/Transformer.ts:83]: To revisit in a future native PR.

**Review by pmconne — APPROVED** [review]

**Review by mgooding — APPROVED** [review]

File: test-apps/imodel-transformer/src/Transformer.ts [hunk 3]
```diff
@@ -129,13 +164,42 @@ export class Transformer extends IModelTransformer {
         statement.bindString("categoryName", categoryName);
         while (DbResult.BE_SQLITE_ROW === statement.step()) {
           const categoryId = statement.getValue(0).getId();
-          this.exporter.excludeElementCategory(categoryId);
-          this.exporter.excludeElement(categoryId);
+          this.exporter.excludeElementCategory(categoryId); // exclude elements in this category
```
[the remark this answers is not in the document]
Comment (scsewall) [reply test-apps/imodel-transformer/src/Transformer.ts]: I agree that `excludeElementsInCategory` is better.

**Review by MichaelBelousov — APPROVED** [review]

**Pull request merged**